File: packages/test-plugin/android/src/main/java/com/capkit/test/TestConfig.kt
package com.capkit.test

import com.getcapacitor.Plugin

/**
 * Helper class to manage the plugin configuration.
 * It parses the values from capacitor.config.json
 */
class TestConfig(plugin: Plugin) {

    val customMessage: String

    init {
        // "customMessage" must match the key in capacitor.config.ts
        customMessage = plugin.getConfig().getString("customMessage") ?: " (from config)"
    }
}

File: packages/test-plugin/ios/Sources/TestPlugin/TestConfig.swift
import Foundation

/**
 * Helper struct to manage the plugin configuration.
 * It parses the values from capacitor.config.json
 */
public struct TestConfig {

    // Define configuration keys for consistency
    private struct Keys {
        static let customMessage = "customMessage"
    }

    // Default values
    private let defaultCustomMessage = " (from config)"

    // Public accessible properties
    public var customMessage: String

    /**
     * Initialize with the config dictionary from the plugin.
     */
    init(config: [AnyHashable: Any]?) {
        let config = config ?? [:]

        // Parse customMessage with a default fallback
        self.customMessage = config[Keys.customMessage] as? String ?? defaultCustomMessage
    }
}
